roachtest: improve the `cluster-init` roachtest

knz opened the pull request:
Fixes #67105
Fixes  #66516.
Fixes  #69949.
Fixes #70003

Release justification: non-production code changes

Release note: None

knz committed:
roachtest: add some clarifying logs to `cluster-init`

Release justification: non-production code changes

Release note: None

knz committed:
roachtest: work around an infra limitation in `cluster-init`

See discussion here;
https://github.com/cockroachdb/cockroach/issues/67105#issuecomment-915566803

Release justification: non-production code changes

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69940)
<!-- Reviewable:end -->

tbg committed:
roachtest: simplify cluster-init by using c.Start

Release justification: test-only change
Release note: None

tbg committed:
roachtest: improve WaitFor3XReplication

Give it a progress bar and a rudimentary context cancellation policy.

Release justification: testing-only changes
Release note: None

tbg reviewed (commented):
I just pushed two commits, one of which makes this test use `c.Start` (the other is a minor cleanup). I verified that the test works using

`TEST=acceptance/cluster-init LOCAL=y SHORT=y COUNT=1 ./pkg/cmd/roachtest/roachstress.sh`

As far as I'm concerned the result is ready for the merge, but since I wrote it perhaps someone else on this review (server or test-eng) wants to sign off officially.

I would also suggest closing https://github.com/cockroachdb/cockroach/issues/67105 (tests should not manage their own CRDB processes, and when they do it's fair that they would wipe the cluster before termination) and also #66516 (since the test is now idiomatic).

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69940#-:-MjThcPKEaLXQWM4KCXo:bz43436)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stevendanna and @tbg)



<!-- Sent from Reviewable.io -->

knz reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69940#-:-MjU7Rd346Z6r-Bhmt2Z:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stevendanna)



<!-- Sent from Reviewable.io -->

tbg commented:
bors r=knz
TFTR!
